tsl/test/sql/bgw_job_ddl.sql

```diff
@@ -40,35 +61,77 @@ RESET ROLE;
 
 -- Test that renaming a user changes keeps the job assigned to that user.
 ALTER USER another_user RENAME TO renamed_user;
-SELECT proc_name, owner FROM _timescaledb_config.bgw_job WHERE id = :job_id;
+SELECT * FROM my_jobs;
+
+-- Test that renaming the procedure also modifies the entry in the
+-- jobs table.
+ALTER PROCEDURE more_magic RENAME TO magic;
+SELECT * FROM my_jobs;
+
+-- Test that modifying the schema also modifies the entry in the jobs
+-- table.
+CREATE SCHEMA frugal;
+ALTER PROCEDURE magic SET SCHEMA frugal;
+ALTER PROCEDURE some_magic SET SCHEMA frugal;
+SELECT * FROM my_jobs;
+
+-- Test that renaming the schema will rename the procedure schema
+START TRANSACTION;
+ALTER SCHEMA frugal RENAME TO wicked;
+SELECT * FROM my_jobs;
+ROLLBACK;
 
 \set VERBOSITY default
 \set ON_ERROR_STOP 0
 
+SELECT * FROM my_jobs;
+
 -- Test that dropping a user owning a job fails.
 DROP USER renamed_user;
-SELECT proc_name, owner FROM _timescaledb_config.bgw_job WHERE id = :job_id;
+
+-- Test that dropping the procedure fails since there is a background
+-- job using it.
+DROP PROCEDURE frugal.magic;
 
 -- Test that re-assigning objects owned by an unknown user still fails
 REASSIGN OWNED BY renamed_user, unknown_user TO :ROLE_DEFAULT_PERM_USER;
-SELECT proc_name, owner FROM _timescaledb_config.bgw_job WHERE id = :job_id;
+
+-- Test that dropping the schema without CASCADE will error out
+DROP SCHEMA frugal;
 
 \set ON_ERROR_STOP 1
 
 -- Test that reassigning the owned job actually changes the owner of
 -- the job.
 START TRANSACTION;
 REASSIGN OWNED BY renamed_user TO :ROLE_DEFAULT_PERM_USER;
-SELECT proc_name, owner FROM _timescaledb_config.bgw_job WHERE id = :job_id;
+SELECT * FROM my_jobs;
 ROLLBACK;
 
 -- Test that reassigning to postgres works
 REASSIGN OWNED BY renamed_user TO :ROLE_CLUSTER_SUPERUSER;
-SELECT proc_name, owner FROM _timescaledb_config.bgw_job WHERE id = :job_id;
+SELECT * FROM my_jobs;
 
 -- Dropping the user now should work.
 DROP USER renamed_user;
 
+-- Dropping using Cascade should work and remove the background worker
+-- entry as well.
```

Comment:
Will it also stop any running bgws?

Reply:
No, any jobs running will continue running until it is done. After that no new jobs will be spawned.